File: 1.Basic-Concepts/i.OOP/go/oop-like.go
package main

import "fmt"

type Rectangle struct{
	length int
	width int
}

// area method definition on receiver r of struct Rectangle
func (r *Rectangle) area() int {
	return r.length * r.width
}

// initialize method definition on receiver r of struct Rectangle
func (r *Rectangle) initialize(length int, width int) {
	r.length = length;
	r.width = width;
}

// initialize method definition on receiver r of struct Rectangle
func (r *Rectangle) changeLength(length int){
	r.length = length;
}

func main(){
	var rec Rectangle;
	
	rec.initialize(20,50);
	fmt.Println("Rec width : ",rec.width, " length : ",rec.length," , AREA : ",rec.area());
	
	rec.changeLength(80);
	fmt.Println("Rec width : ",rec.width, " length : ",rec.length," , AREA : ",rec.area());
}

File: 1.Basic-Concepts/i.OOP/oop-like.c
#include <stdio.h>
#include <stdlib.h>

int area(struct Rectangle rec); 
void initialize(struct Rectangle *rec, int length, int width); // Pass by address
void changeLength(struct Rectangle *rec, int length); 

struct Rectangle 
{
    int length;
    int width;
};

void initialize(struct Rectangle *rec, int length, int width)
{
    rec->length = length;
    rec->width = width;
}

void changeLength(struct Rectangle *rec, int length)
{
    rec->length = length;
}

int area(struct Rectangle rec)
{
  return rec.length * rec.width;
}

int main()
{
    struct Rectangle rec;
 
    initialize(&rec,10,20);
    printf("After initialize(&rec,10,2) fn call : rec width : %d, length : %d\n",rec.width,rec.width );

    printf("Area : %d \n",area(rec));
    
    changeLength(&rec, 100);
    
    printf("Area : %d \n",area(rec));

    return 0;
}